pkg/sources/github/github.go

```diff
@@ -377,11 +377,16 @@ func (s *Source) enumerate(ctx context.Context, apiEndpoint string) (*github.Cli
 	return installationClient, nil
 }
 
-func (s *Source) enumerateBasicAuth(ctx context.Context, basicAuth *credentialspb.BasicAuth) error {
-	s.apiClient = github.NewClient(&http.Client{Transport: &github.BasicAuthTransport{
+func (s *Source) enumerateBasicAuth(ctx context.Context, apiEndpoint string, basicAuth *credentialspb.BasicAuth) error {
+	httpClient := &http.Client{Transport: &github.BasicAuthTransport{
```

Comment:
Although it isn't in the current code, I think it would be better to use `s.httpClient` and modify the transport. It was set up earlier as a retriable client, so it would be good to continue to use that.